### PR TITLE
functions: add cloudbuild.yaml example

### PR DESCRIPTION
**Pull request by tbpg**
Will be used on https://cloud.google.com/functions/docs/testing/test-cicd#functions-testing-frameworks-go.

### CODE REVIEW

File: functions/ci_cd/cloudbuild.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# [START functions_ci_cd_cloud_build]
+steps:
+  - name: 'mirror.gcr.io/library/golang'
+    args: ['go', 'version']
+  - name: 'mirror.gcr.io/library/golang'
+    args: ['go', 'test']
+    dir: 'functions/autodeploy'
+  - name: 'gcr.io/cloud-builders/gcloud'
+    args: ['functions', 'deploy', '[YOUR_DEPLOYED_FUNCTION_NAME]', '[YOUR_FUNCTION_TRIGGER]', '--runtime', '[YOUR_RUNTIME]', '--entry-point', '[YOUR_FUNCTION_NAME_IN_CODE]']
```
Comment (grant) [inline functions/ci_cd/cloudbuild.yaml]: Can't we add  the runtime and other args? I assume we are using `go111`?

**Comment by grant** [comment]
Ah, this PR looks related to the friction log you wrote.
https://cloud.google.com/functions/docs/testing/test-cicd#functions-testing-frameworks-go

**Comment by grant** [comment]
I suggest we add a region tag for this yaml: `functions_ci_cd_cloud_build_go`

File: functions/ci_cd/cloudbuild.yaml (hunk 1 above)
Comment (tbpg) [reply functions/ci_cd/cloudbuild.yaml]: Following the existing example. Would be nice not to have to update it for each runtime version as we update. Can update if you feel strongly.

We could also rework this sample at a later date, doing Node.js and Go (and others) at the same time.

**Pull request merged**